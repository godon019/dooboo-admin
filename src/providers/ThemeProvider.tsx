import React, { useContext } from 'react';
import { ThemeType, createTheme } from '../theme';

import { AppContext } from '../contexts';
import { ThemeProvider as StyledComponentThemeProvider } from 'styled-components';

interface Props {
  children?: any;
}
const ThemeProvider = (props: Props) => {
  const fromAppProvider = useContext(AppContext);
  /**
   * StyledComponentThemeProvider is normally used with AppProvider as its parent
   * and get `theme` value from AppProvider
   * if AppProvider doesn't exist
   * then use ThemeType.LIGHT as default `theme`
   */
  const theme = fromAppProvider ? fromAppProvider.state.theme : ThemeType.LIGHT;
  return (
    <StyledComponentThemeProvider theme={createTheme(theme)}>
      {props.children}
    </StyledComponentThemeProvider>
  );
};

export default ThemeProvider;
