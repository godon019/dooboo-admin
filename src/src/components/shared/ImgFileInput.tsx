import React, {
  FormEventHandler,
  MouseEventHandler,
  SyntheticEvent,
  useRef,
} from 'react';
import styled, { CSSProp } from 'styled-components';

import Button from './Button';
import { ButtonPrimaryLight } from '../ui/Buttons';
import { useImgUpload } from './ServiceModal/imageControl';

const CustomFileInput = styled.div`
  display: flex;
  width: 100%;
  border: 1px solid #d9d9da;
  padding: 0.7rem;
  box-sizing: border-box;
  background-color: #fff;
`;
const FileInputDom = styled.input`
  display: none;
`;
const ContentView = styled.div`
  display: flex;
  align-items: center;
  flex: 4;
`;
const UploadBtnView = styled.div`
  flex: 1;
`;
const PreviewMsg = styled.span`
  font-size: 0.8rem;
`;
const PreviewName = styled.span`
  margin: 0 0.8rem;
  font-size: 0.8rem;
`;
const PreviewImg = styled.img`
  width: 10%;
  height: 100%;
`;
const FileInputArea = styled.div`
  display: flex;
  flex: 3;
  flex-direction: column;
  justify-content: center;
`;

type FileInputProps = {
  name: string;
  id: string;
  previewImg?: any;
  previewName?: string;
  readOnly?: boolean;
  customCss?: CSSProp;
  onChange?: FormEventHandler;
};

const FileInput: React.FC<
  FileInputProps & React.HTMLAttributes<HTMLDivElement>
> = ({ name, id, previewImg, previewName, readOnly, customCss, onChange }) => {
  const fileInputRef = useRef(null);
  const triggerFileInput = () => {
    if (fileInputRef) {
      fileInputRef.current.click();
    }
  };
  return (
    <CustomFileInput>
      <ContentView>
        {previewImg ? (
          <PreviewImg src={previewImg} data-testid={`${name}-preview`} />
        ) : (
          <PreviewMsg data-testid={'defaultPreview'}>
            이미지를 첨부해주세요. ( 형식 : jpg, png, gif | 크기 : 512x512 )
          </PreviewMsg>
        )}
        {previewName ? <PreviewName>{previewName}</PreviewName> : null}
        <FileInputDom
          data-testid={`${name}-file`}
          type='file'
          name={name}
          id={id}
          ref={fileInputRef}
          onChange={onChange}
        />
      </ContentView>
      <UploadBtnView>
        <Button
          data-testid={`${name}-upload-btn`}
          text={'이미지 첨부'}
          outline={true}
          customStyle={{ marginBottom: 0, height: null, padding: '.7rem' }}
          onClick={triggerFileInput}
        />
      </UploadBtnView>
    </CustomFileInput>
  );
};

export default FileInput;
