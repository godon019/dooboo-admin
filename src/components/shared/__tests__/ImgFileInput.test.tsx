import { fireEvent, getByTestId, render } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import ImgFileInput from '../ImgFileInput';
import React from 'react';
import ThemeProvider from '../../../providers/ThemeProvider';

let props: any;
let component: React.ReactElement;
let testingLib: any;

describe('[ImgFileInput] render', () => {
  beforeEach(() => {
    props = {
      history: {
        goBack: jest.fn(),
      },
    };
    component = (
      <ThemeProvider>
        <ImgFileInput name={'imgfile-name'} id={'imgfile-id'} />
      </ThemeProvider>
    );
  });

  it('renders without crashing', () => {
    const rendered: ReactTestRendererJSON = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      // fireEvent(testingLib.getByTestId('btn'), 'click');
      // expect(cnt).toBe(2);
    });
  });
});
