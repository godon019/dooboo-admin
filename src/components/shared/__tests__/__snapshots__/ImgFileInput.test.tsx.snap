// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ImgFileInput] render renders without crashing 1`] = `
<div
  className="sc-bZQynM jwgxKa"
>
  <div
    className="sc-htoDjs iKOCRY"
  >
    <span
      className="sc-iwsKbI bGxQVD"
      data-testid="defaultPreview"
    >
      이미지를 첨부해주세요. ( 형식 : jpg, png, gif | 크기 : 512x512 )
    </span>
    <input
      className="sc-gzVnrw aZFdW"
      data-testid="imgfile-name-file"
      id="imgfile-id"
      name="imgfile-name"
      type="file"
    />
  </div>
  <div
    className="sc-dnqmqq kHYyxp"
  >
    <button
      className="sc-bdVaJa bEdxjs"
      data-testid="imgfile-name-upload-btn"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#069ccd",
          "height": null,
          "marginBottom": 0,
          "padding": ".7rem",
        }
      }
      type="button"
    >
      <div>
        <span
          className="sc-bxivhb koiFyU"
          style={
            Object {
              "color": "#069ccd",
            }
          }
        >
          이미지 첨부
        </span>
      </div>
    </button>
  </div>
</div>
`;
